Fix thermal manager setpoint tracking tests

## Summary
2 tests are failing in the thermal manager related to original setpoint tracking.

## Affected Tests
- `test_tracks_original_setpoint` - `_original_setpoints` dict is empty when it should contain the tracked setpoint
- `test_clears_tracking_when_offset_zero` - `_original_setpoints` not being cleared when offset is zero

## Error Messages
```
test_tracks_original_setpoint:
AssertionError: assert 'climate.living_room' in {}
  where {} = ThermalManager._original_setpoints

test_clears_tracking_when_offset_zero:
AssertionError: assert 1 == 0
  where 1 = len({'climate.living_room': 24.0})
```

## Root Cause
The `_original_setpoints` tracking logic in `ThermalManager` appears to have changed:
1. Setpoints are not being stored when applying temperature offsets
2. Setpoints are not being cleared when offset returns to zero

## Investigation Needed
- Review `apply_climate_control` method in thermal_manager.py
- Check if the tracking logic was intentionally removed or modified
- Update tests or fix the implementation

## Files to Review
- `tests/test_thermal_manager.py`
- `custom_components/localshift/thermal_manager.py`

## Priority
Medium - affects thermal management functionality

--- a/tests/test_thermal_manager.py
+++ b/tests/test_thermal_manager.py
@@ -665,6 +665,7 @@ class TestClimateControl:
     async def test_tracks_original_setpoint(self, thermal_manager, coordinator_data):
         """Climate control tracks original setpoint to prevent ratchet."""
         thermal_manager._get_switch_state = lambda k: k == "thermal_management_enabled"
+        coordinator_data.daily_thermal_mode = ThermalMode.COOL
         coordinator_data.climate_control_entities = ["climate.living_room"]
         coordinator_data.climate_states = {
             "climate.living_room": {
@@ -700,6 +701,7 @@ class TestClimateControl:
         thermal_manager._get_switch_state = lambda k: k == "thermal_management_enabled"
         thermal_manager._original_setpoints = {"climate.living_room": 24.0}
         thermal_manager._current_active_offset = -2.0
+        coordinator_data.daily_thermal_mode = ThermalMode.COOL
         coordinator_data.climate_control_entities = ["climate.living_room"]
 
         await thermal_manager.async_apply_climate_control(

